drivers/periph_common/spi.c

```diff
@@ -25,7 +25,7 @@
 #include "cpu.h"
 #include "periph/spi.h"
 
-#ifdef SPI_NUMOF
+#if defined(SPI_NUMOF) && !defined(MODULE_SOFT_SPI_AS_PERIPH_SPI)
```

Comment:
I wonder if this new soft spi module can coexist with hardware SPI

Reply:
You can still use `soft_spi` together with `periph_spi`. But `soft_spi_as_periph_spi` hooks up `soft_spi` into the `periph_spi` interface and, hence, is incompatible.

Reply:
I wonder if some day an approach like done on #14602 could be used for soft SPI/UART, so if you have used all the SPI/UARTs of the device you can always have the option to enable the bitbanged versions of both protocols.